core/state_processor.go

```diff
@@ -178,7 +178,7 @@ func ApplyTransaction(config *params.ChainConfig, bc ChainContext, author *commo
 
 func InsertBlockHashHistoryAtEip2935Fork(statedb *state.StateDB, prevNumber uint64, prevHash common.Hash, chain consensus.ChainHeaderReader) {
 	// Make sure that the historical contract is added to the witness
-	statedb.Witness().TouchFullAccount(params.HistoryStorageAddress[:], true, math.MaxUint64)
+	statedb.Witness().TouchFullAccount(params.HistoryStorageAddress[:], true, true, math.MaxUint64)
```

Comment:
right this is only set to `true` so that if someone, later, writes to the contract (which should not happen :tm: , see comment above) then they don't pay for the fill costs.